File: 0145-binary-tree-postorder-traversal/0145-binary-tree-postorder-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> postorderTraversal(TreeNode* root) {
        vector<int> myVector;
        traverse(root,myVector);
        return myVector;
    }

    void traverse(TreeNode* root, vector<int>& myVector){
        if(root==nullptr){
            return;
        }
        if(root->left == nullptr && root->right == nullptr){
            myVector.emplace_back(root->val);
            return;
        }
        traverse(root->left,myVector);
        traverse(root->right,myVector);
        myVector.emplace_back(root->val);
    }
};

File: 0145-binary-tree-postorder-traversal/0145-binary-tree-postorder-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> postorderTraversal(TreeNode root) {
        List<Integer> myList = new ArrayList<Integer>();
        traverse(root,myList);
        return myList;
    }

    public void traverse(TreeNode root, List<Integer> myList){
        if(root == null){
            return;
        }
        if(root.left==null && root.right == null){
            myList.add(root.val);
            return;
        }
        traverse(root.left,myList);
        traverse(root.right,myList);
        myList.add(root.val);
    }
}